### PR TITLE
generator: Upgrade APC to 4.2.8

### PR DESCRIPTION
**Pull request by daenney**

### CODE REVIEW

File: snmp.yml [hunk 1]
```diff
@@ -1118,6 +1141,66 @@ apcups:
...
+  - name: upsHighPrecBatteryPackOnlyTemperature
+    oid: 1.3.6.1.4.1.318.1.1.1.2.3.10.4.1.4
+    type: gauge
+    help: The battery pack temperature measured in 10ths of degree Celcius - 1.3.6.1.4.1.318.1.1.1.2.3.10.4.1.4
+    indexes:
+    - labelname: upsHighPrecBatteryPackOnlyIndex
+      type: gauge
+  - name: upsHighPrecBatteryPackOnlyStatus
+    oid: 1.3.6.1.4.1.318.1.1.1.2.3.10.4.1.5
+    type: OctetString
```
Comment (daenney) [inline snmp.yml]: Wondering whether this should be overridden to `DisplayString`

File: snmp.yml [hunk 2]
```diff
@@ -1118,6 +1141,66 @@ apcups:
...
+    help: The battery status for the pack only - 1.3.6.1.4.1.318.1.1.1.2.3.10.4.1.5
+    indexes:
+    - labelname: upsHighPrecBatteryPackOnlyIndex
+      type: gauge
+  - name: upsHighPrecBatteryPackOnlyHealth
+    oid: 1.3.6.1.4.1.318.1.1.1.2.3.10.4.1.6
+    type: OctetString
```
Comment (daenney) [inline snmp.yml]: Wondering whether this should be overridden to `DisplayString`

File: snmp.yml (hunk 1 above)
Comment (brian-brazil) [reply snmp.yml]: Is it always an ASCII/UTF-8 string?

Comment (daenney) [reply snmp.yml]: Mmm, according to the MIB docs:

```
The battery status for the pack only.
bit 0 Disconnected
bit 1 Overvoltage
bit 2 NeedsReplacement
bit 3 OvertemperatureCritical
bit 4 Charger
bit 5 TemperatureSensor
bit 6 BusSoftStart
bit 7 OvertemperatureWarning
```

File: snmp.yml (hunk 2 above)
Comment (daenney) [reply snmp.yml]: According to the MIB docs:

```
The battery pack health.
bit 0 Battery lifetime okay
bit 1 Battery lifetime near end, order replacement pack
bit 2 Battery lifetime exceeded, replace battery
bit 3 Battery lifetime near end acknowledged, order replacement pack
bit 4 Battery lifetime exceeded acknowledged, replace battery
bit 5 Battery measured lifetime near end, order replacement pack
bit 6 Battery measured lifetime near end acknowledged, order replacement pack
```

File: snmp.yml (hunk 1 above)
Comment (SuperQ) [reply snmp.yml]: That seems like it returns a single byte and you have to look at the bits yourself.

Comment (daenney) [reply snmp.yml]: Yeah. I guess so. So we leave it as such?

**Comment by SuperQ** [comment]
`upsAdvBatteryChargerStatus` would make a good StateSet.

File: snmp.yml (hunk 1 above)
Comment (SuperQ) [reply snmp.yml]: Yea, I would leave it, it doesn't seem like a string, and without a device to test with it's hard to say.

**Comment by daenney** [comment]
Yap. Added.

File: snmp.yml (hunk 1 above)
Comment (brian-brazil) [reply snmp.yml]: You might be able to do something with regex extracts.

**Pull request merged**